test/types/reply.test-d.ts

```diff
@@ -32,6 +34,11 @@ const getHandler: RouteHandlerMethod = function (_request, reply) {
   expectType<(payload: any) => string | ArrayBuffer | Buffer>(reply.serialize)
   expectType<(fulfilled: () => void, rejected: (err: Error) => void) => void>(reply.then)
   expectType<FastifyInstance>(reply.server)
+  expectAssignable<((httpStatus: string) => DefaultSerializationFunction)>(reply.getSerializationFunction)
+  expectAssignable<((schema: {[key: string]: unknown}) => DefaultSerializationFunction)>(reply.getSerializationFunction)
+  expectAssignable<((schema: {[key: string]: unknown}, httpStatus?: string) => DefaultSerializationFunction)>(reply.compileSerializationSchema)
+  expectAssignable<((input: {[key: string]: unknown}, schema: {[key: string]: unknown}, httpStatus?: string) => unknown)>(reply.serialize)
+  expectAssignable<((input: {[key: string]: unknown}, httpStatus: string) => unknown)>(reply.serialize)
```

Comment:
`reply.serialize` accepts a payload [only](https://github.com/fastify/fastify/blob/main/docs/Reference/Reply.md#introduction)

Reply:
Sorry, I think I don't fully get you. Do you refer to `serialzeInput` or `serialize`? 🤔

Reply:
Yes, it refers to the second one, `serializeInput` 😅 
I can add a test for `serialize` if you want

Reply:
I have submitted the suggestions to explain myself